### PR TITLE
 Timeout for Inprocess Executor on Windows

### PR DESCRIPTION
**Pull request by tokatoka**
Windows timeout using this win32 API
https://docs.microsoft.com/en-us/windows/win32/api/threadpoollegacyapiset/

but remove_timeout() inside crash handler is not implemented for windows (#224)
on windows this is not as simple as on unix :<

### CODE REVIEW

**Comment by domenukk** [comment]
This is cool!

File: libafl/src/executors/inprocess.rs [hunk 1]
```diff
@@ -621,6 +705,8 @@ mod windows_exception_handler {
         I: Input,
         Z: HasObjective<I, OF, S>,
     {
+        // TODO windows_remove_timeout()
+        // We can use DeleteTimerQueueEx, but you need a pointer to the timer_queue, which is not accessible from inproc_crash_handler.
```
Comment (domenukk) [inline libafl/src/executors/inprocess.rs:709]: Can we make a somewhat hackish solution like storing the pointer thread local?

File: libafl/src/executors/timeout.rs [hunk 2]
```diff
@@ -82,16 +98,70 @@ impl<E> TimeoutExecutor<E> {
     }
 
     #[cfg(windows)]
-    pub fn new(executor: E, exec_tmout: Duration) -> Self {
-        Self { executor }
+    pub unsafe fn new(executor: E, exec_tmout: Duration) -> Self {
+        let milli_sec = exec_tmout.as_millis() as u32;
```
Comment (domenukk) [inline libafl/src/executors/timeout.rs:102]: Why should `new` be unsafe?

File: libafl/src/executors/inprocess.rs (hunk 1 above)
Comment (tokatoka) [reply libafl/src/executors/inprocess.rs:709]: Yes, I guess it's possible, but I'm undecided on which pointer to store. 
1. pointer to the `executor: Executor` or
2. pointer to the `timer_queue: HANLDE` 
3. or other thingy?

IMO, the pointer to the executor seems better, if we can tie `remove_timeout()` to the `Exectuor` then we can use like `executor->remove_timeout()` inside crash_handler

File: libafl/src/executors/timeout.rs (hunk 2 above)
Comment (tokatoka) [reply libafl/src/executors/timeout.rs:102]: no need to be `unsafe`
I forgot to delete it 😅

File: libafl/src/executors/inprocess.rs (hunk 1 above)
Comment (domenukk) [reply libafl/src/executors/inprocess.rs:709]: I'd probably just do the `HANDLE`, if it's thread local you usually don't have more than a single pointer at this point. Buit either way is fine ;)

Comment (tokatoka) [reply libafl/src/executors/inprocess.rs:709]: I haven't tried, but the microsoft doc says the callback is run in another thread pool, so thread_local! might not work.
I will try passing the ptr to the executor to the callback :)

**Comment by tokatoka** [comment]
Well I made it by adding an extra timer_queue pointer to the InProcessHandlerData on windows

**Pull request merged**